hadoop-hdfs-project/hadoop-hdfs-rbf/src/test/java/org/apache/hadoop/hdfs/server/federation/router/TestRouterTrash.java

```diff
@@ -144,6 +144,40 @@ private boolean addMountTable(final MountTable entry) throws IOException {
     return addResponse.getStatus();
   }
 
+  @Test
+  public void testMoveToTrashAutoCreateUserHome() throws IOException, URISyntaxException, InterruptedException {
```

Comment:
CAn you create a negative test case: where the user directory already exists we don't create it.